numpy/doc/structured_arrays.py

```diff
@@ -6,231 +6,430 @@
-equal sized list of values. The format list contains any type/shape specifier
-allowed in other contexts. The names must be strings. There are two optional
-keys: 'offsets' and 'titles'. Each must be a correspondingly matching list to
-the required two where offsets contain integer offsets for each field, and
-titles are objects containing metadata for each field (these do not have
-to be strings), where the value of None is permitted. As an example: ::
-
- >>> x = np.zeros(3, dtype={'names':['col1', 'col2'], 'formats':['i4','f4']})
- >>> x
- array([(0, 0.0), (0, 0.0), (0, 0.0)],
-       dtype=[('col1', '>i4'), ('col2', '>f4')])
+In this shorthand notation any of the :ref:`string dtype specifications
+<arrays.dtypes.constructing>` may be used in a string, separated by commas. The
+itemsize and byte offsets of the fields are determined automatically, and the
+field names are given the default names ``f0``, ``f1``, etc. ::
 
-The other dictionary form permitted is a dictionary of name keys with tuple
-values specifying type, offset, and an optional title. ::
+ >>> np.dtype('i8,f4,S3')
+ dtype([('f0', '<i8'), ('f1', '<f4'), ('f2', 'S3')])
+ >>> np.dtype('3int8, float32, (2,3)float64')
+ dtype([('f0', '|i1', 3), ('f1', '>f4'), ('f2', '>f8', (2, 3))])
 
- >>> x = np.zeros(3, dtype={'col1':('i1',0,'title 1'), 'col2':('f4',1,'title 2')})
- >>> x
- array([(0, 0.0), (0, 0.0), (0, 0.0)],
-       dtype=[(('title 1', 'col1'), '|i1'), (('title 2', 'col2'), '>f4')])
+3. A dictionary of field parameters
+`````````````````````````````````
 
-Accessing and modifying field names
-===================================
+This is the most flexible form of specification since it allows control
+over the byte-offsets of the fields and the itemsize of the structure.
 
-The field names are an attribute of the dtype object defining the structure.
-For the last example: ::
+The dictionary has two required keys, 'names' and 'formats', and four optional
+keys, 'offsets', 'itemsize', 'aligned' and 'titles'. 'names' and 'formats'
+should respectively correspond to a list of field names and a list of dtype
+specifications of the same length. The optional 'offsets' key must correspond
+to a list of integer byte-offsets of each field within the structure, of the
+same length. If 'offsets' is not given the offsets are determined
+automatically. The optional 'itemsize' key should correspond to an integer
+describing the total size in bytes of the dtype, which must be large enough
+that all the fields are contained. ::
 
- >>> x.dtype.names
- ('col1', 'col2')
- >>> x.dtype.names = ('x', 'y')
- >>> x
+ >>> np.zeros(3, dtype={'names': ['col1', 'col2'], 'formats': ['i4','f4']})
+ array([(0, 0.0), (0, 0.0), (0, 0.0)],
+       dtype=[('col1', '>i4'), ('col2', '>f4')])
+ >>> np.zeros(3, dtype={'names':    ['col1', 'col2'],
+ ...                    'formats':  ['i4','f4'],
+ ...                    'offsets':  [0, 4],
+ ...                    'itemsize': 12})
```

Comment:
Aligning the `[...]`  isn't PEP8. It may take a bit of getting used to but eventually the unaligned versions become easier to read.